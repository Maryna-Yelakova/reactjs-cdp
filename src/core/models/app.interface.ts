export interface AppState {
      query:string;
      list:Array<MovieModel>;
      movie: MovieModel;
}

export interface MovieModel {
      id: number;
      title: string;
      tagline: string;
      vote_average: number;
      vote_count: number;
      release_date: string;
      poster_path: string;
      overview: string;
      budget: number;
      revenue: number;
      runtime: number;
      genres:Array<string>;
}
