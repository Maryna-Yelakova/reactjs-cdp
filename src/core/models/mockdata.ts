export const MockMovie = {
      id: 0,
      title: '',
      tagline: '',
      vote_average: 0,
      vote_count: 0,
      release_date: '',
      poster_path: '',
      overview: '',
      budget: 0,
      revenue: 0,
      runtime: 0,
      genres: ['']
}

export const MockMovieList = [
  {
    id: 1,
    title: 'Imperial concubine',
    tagline: 'comedy',
    vote_average: 2045,
    vote_count: 2045,
    release_date: '',
    poster_path: '',
    overview: '',
    budget: 0,
    revenue: 0,
    runtime: 0,
    genres: ['']
  },
  {
    id: 2,
    title: 'Meng Qi Shi Shen',
    tagline: 'romantic',
    vote_average: 764,
    vote_count: 764,
    release_date: '',
    poster_path: '',
    overview: '',
    budget: 0,
    revenue: 0,
    runtime: 0,
    genres: ['']
  }
]
