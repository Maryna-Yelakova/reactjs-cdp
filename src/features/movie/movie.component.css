.movie {
	display: flex;
	box-sizing: border-box;
	width: 100%;
	padding: 40px;
	margin-bottom: 20px;
}

.movie_info {
	padding: 0 40px;
}

.movie_name {
	font-size: 30px;
	line-height: 40px;
	font-weight: bold;
	color: #e83e63;
}

.movie_description {
	padding: 40px 0;
}
