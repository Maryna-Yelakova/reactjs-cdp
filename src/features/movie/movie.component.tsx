import * as React from 'react';
import { connect } from 'react-redux';

import './movie.component.css';
// models
import { MovieModel } from '../../core/models/app.interface';

// images
import * as posterImage from '../../assets/images/poster1.jpg';

export const Movie: React.SFC<MovieModel> = (movie) => {

  return (
    <div className="movie">
      <img
        src={ posterImage }
        width="200"
        height="302"
        alt="picture"
      />
      <div className="movie_info">
        <div className="movie_main-info" data-cy="info">
          <h2 className="movie_name">{movie.title}</h2>
          <p className="movie_genre">{movie.vote_average}</p>
        </div>
        <p className="movie_description">{movie.overview}</p>
      </div>
    </div>
  );

}
