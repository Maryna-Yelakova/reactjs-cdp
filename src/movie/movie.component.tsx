import * as React from 'react';
import './movie.component.css';

export const Movie = ({movie}) => {
  return (
          <div className="movie">
             <h2>{movie.title}</h2>
             <span>{movie.vote_average}</span>
          </div>
          );
}
