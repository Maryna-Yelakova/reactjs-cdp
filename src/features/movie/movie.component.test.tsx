import * as  React from 'react';
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Movie } from './movie.component';
import { MockMovie } from '../../core/models/mockdata';


describe('Movie', () => {
 it('should render correctly', () => {
   const component = shallow(<Movie {...MockMovie} />);

   expect(component).toBeDefined();
   expect(shallowToJson(component)).toMatchSnapshot();
 });

});
