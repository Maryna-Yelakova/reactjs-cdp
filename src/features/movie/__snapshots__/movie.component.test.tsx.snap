// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render correctly 1`] = `
<div
  className="movie"
>
  <img
    alt="picture"
    height="302"
    src={
      Object {
        "default": "test-file-stub",
      }
    }
    width="200"
  />
  <div
    className="movie_info"
  >
    <div
      className="movie_main-info"
    >
      <h2
        className="movie_name"
      />
      <p
        className="movie_genre"
      >
        0
      </p>
    </div>
    <p
      className="movie_description"
    >
      Some mock overview
    </p>
  </div>
</div>
`;
